memo/managers.py

```diff
@@ -73,3 +74,63 @@ def get_card_amount_in_deck(self, deck):
 class DeckManager(models.Manager):
     def get_decks_by_owner(self, user):
         return self.filter(owner=user)
+
+
+class CardAnswerHistoryManager(models.Manager):
+    def get_consecutive_days(self, user):
+        from .models import Card
+
+        max_daily_date_times = (
+            self.filter(owner=user)
+            .extra(select={"day": "date( created_at )"})
+            .values("day")
+            .annotate(latest_datetime=Max("created_at"))
+        )
+
+        objects = (
+            self.filter(
+                created_at__in=[
+                    entry["latest_datetime"] for entry in max_daily_date_times
+                ]
+            )
+            .values("card", "created_at")
+            .order_by("-created_at")
+        )
+
+        result = []
+
+        if objects:
```

Comment:
You don't need to check if the `objects` has something. If you try to iterate over it when it's empty, python will actually just ignore it and just move on.
```suggestion
```

Reply:
This conditional is needed because if a new user doesn't have any answer history, it'll return an error in the line ` objects[0].get("created_at")`, because it will try to access an index without existing.

Reply:
Oh yeah, that makes sense